x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/rules_stats_collector.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/core/server';
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import type { CspmRulesStats } from './types';
+import { LATEST_FINDINGS_INDEX_DEFAULT_NS } from '../../../../common/constants';
+
+interface DocCount {
+  doc_count: number;
+}
+
+interface BenchmarkName {
+  metrics: { 'rule.benchmark.name': string };
+}
+
+interface BenchmarkId {
+  metrics: { 'rule.benchmark.id': string };
+}
+
+interface BenchmarkVersion {
+  metrics: { 'rule.benchmark.version': string };
+}
+interface RuleName {
+  metrics: { 'rule.name': string };
+}
+interface RuleSection {
+  metrics: { 'rule.section': string };
+}
+interface RuleVersion {
+  metrics: { 'rule.version': string };
+}
+interface RuleNumber {
+  metrics: { 'rule.benchmark.rule_number': string };
+}
+interface PostureType {
+  metrics: { 'rule.benchmark.posture_type': string };
+}
+
+interface RulesStats {
+  rules: {
+    buckets: RuleEntity[];
+  };
+}
+
+interface RuleEntity {
+  key: string; // rule_id
+  rule_name: { top: RuleName[] };
+  rule_section: { top: RuleSection[] };
+  rule_version: { top: RuleVersion[] };
+  rule_number: { top: RuleNumber[] };
+  posture_type: { top: PostureType[] };
+  benchmark_id: { top: BenchmarkId[] };
+  benchmark_name: { top: BenchmarkName[] };
+  benchmark_version: { top: BenchmarkVersion[] };
+  passed_findings_count: DocCount;
+  failed_findings_count: DocCount;
+}
+
+const getRulesStatsQuery = (index: string): SearchRequest => ({
+  index,
+  query: {
+    match_all: {},
+  },
+  aggs: {
+    rules: {
+      terms: {
+        field: 'rule.id',
+        order: {
+          _count: 'desc',
+        },
+        size: 100,
+      },
+      aggs: {
+        rule_name: {
+          top_metrics: {
+            metrics: {
+              field: 'rule.name',
+            },
+            size: 1,
```

Comment:
what does this do? i think a `100` would return max 100 results, but not sure i understand why inner aggregations need this value?